### PR TITLE
add map locale function

### PR DESCRIPTION
**Pull request by MohamedTaha880**

### CODE REVIEW

File: src/modules/localization/classes/Common/Localization.class.php [hunk 1]
```diff
@@ -211,4 +211,18 @@ public static function translatableText(Environment $env, $text, $tag = NULL, $l
     $output_text = self::t($output_text,$replace);
     return $output_text; 
   }
+
+  public static function mapLocale($lang){
+     // Define a mapping from short locale codes to full locale strings
+     $localeMap = [
+      'en' => 'en_US',
```
Comment (Aldus83) [inline src/modules/localization/classes/Common/Localization.class.php]: @MohamedTaha880 this hardcoded languages is a no-go.

languages are already present in _languages folder as nodes

just add a field into them to contain full locale
i.e. inside db/_languages/en/data.json you will have {... "locale_code":"en_US" ... }

**Review by Aldus83 — CHANGES_REQUESTED** [review]

File: src/modules/localization/classes/Common/Localization.class.php (hunk 1 above)
Comment (MohamedTaha880) [reply src/modules/localization/classes/Common/Localization.class.php]: @Aldus83 Done

File: src/modules/localization/classes/Common/Localization.class.php [hunk 2]
```diff
@@ -211,4 +211,13 @@ public static function translatableText(Environment $env, $text, $tag = NULL, $l
     $output_text = self::t($output_text,$replace);
     return $output_text; 
   }
+
+  public static function mapLocale($env,$lang){
+    $node = NodeFactory::load($env, $lang);
+    $locale_code = 'it_IT';
```
Comment (Aldus83) [inline src/modules/localization/classes/Common/Localization.class.php:217]: default should not be it but  current language

**Review by Aldus83 — CHANGES_REQUESTED** [review]

File: src/modules/localization/classes/Common/Localization.class.php (hunk 2 above)
Comment (MohamedTaha880) [reply src/modules/localization/classes/Common/Localization.class.php:217]: @Aldus83 
the current language will be (it, en ....) not full locale code
So I put it_IT as default code if we can't find the locale code of the selected language

**Pull request merged**